src/features/campaign-sheet/components/GMInfo.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { StoredCampaign } from "../../../types/Campaign.type";
+import { getDoc } from "firebase/firestore";
+import { getUsersDoc } from "../../../lib/firebase.lib";
+import { UserDocument } from "../../../types/User.type";
+import Avatar from "@mui/material/Avatar";
+import Typography from "@mui/material/Typography";
+import Stack from "@mui/material/Stack";
+
+interface GMInfoProps {
+  campaign: StoredCampaign;
+}
+
+const GMInfo = ({ campaign }: GMInfoProps) => {
+  const [gm, setGm] = useState<UserDocument>({
+    displayName: "",
+    photoURL: "",
+  });
+
+  const gmId = campaign.gmId as string;
+
+  const fetchGM = async () => {
+    const gmSnap = await getDoc(getUsersDoc(gmId));
+    const gmData = gmSnap.data() as UserDocument;
+    setGm(gmData);
+  };
+
+  useEffect(() => {
+    fetchGM();
+  }, []);
```

Comment:
Thanks for the heads up! I definitely should look into useEffect more. I've also never used useCallback before but I'm starting to understand it based on this situation. 

What about something like this? 

```jsx
const GMInfo = ({ gmId }: GMInfoProps) => {
  const [gm, setGm] = useState<UserDocument>();

  useEffect(() => {
    getDoc(getUsersDoc(gmId)).then((gmSnap) => {
      setGm(gmSnap.data());
    });
  }, [gmId]);

  return (
    <Stack
      direction="row"
      sx={{
        m: 2,
        alignItems: "center",
      }}
      spacing={2}
    >
      <Avatar alt={gm?.displayName} src={gm?.photoURL} />
      <Typography variant="h6">{gm?.displayName}</Typography>
    </Stack>
  );
};
```

We don't really need the whole `campaign` in this, just the `gmId`, which is guaranteed to exist based on the conditional in `CampaignSheetPage.tsx`. And if move the fetching directly into useEffect, we don't have to worry about function definitions... right? 

This also addresses your comments on setting the initial state as undefined. I added a few `?` at the end to make Typescript happy. Is there a better way of handling that?

Reply:
Almost perfect - Instead of doing the `gm?.displayName`, I might add a line above your current return that says:
``` typescript
if(!gm) {
  return null;
}
```
The reason being that if we render the Avatar and Typography even without content, they probably still take space up on the page.

Reply:
But yeah I like this approach a lot better